### DISCUSSION TITLE
Some minor bugs with the Black Gate Keyring mod

### DISCUSSION THREAD

**Discussion by Zhelkas** [exult/exult#376] [General]
Hey all,

I just finished a playthrough of U7:BG with the Black Gate Keyring mod, with Exult 1.9.0.  I used the Avatar Pack and had the font changed so it was easier to read.  There were a few minor bugs I noticed, but nothing game-breaking:

-Alagner doesn't die.  The Guardian taunts you with "Thank you for the notebook" and the crystal ball works as it should, showing him getting attacked, but there's no dismembered corpse in his house and he goes about his daily business just fine at the end.

-In the Yew jail, Goth the guard sleeps on the empty cot in the prison cell.  Gharl the imprisoned troll sleeps in Goth's bed.

-After getting in a fight, sometimes the blood trails from the Avatar go on a **long** time, making a huge bloody mess everywhere, even if I fully heal myself.

Just making everyone aware of these issues.  Please let me know if any further information would be helpful.  Thanks!

**Comment by DominusExult** [comment]
> -Alagner doesn't die. The Guardian taunts you with "Thank you for the notebook" and the crystal ball works as it should, showing him getting attacked, but there's no dismembered corpse in his house and he goes about his daily business just fine at the end.

Do you have a savegame, preferably before passing on the notebook?

> -In the Yew jail, Goth the guard sleeps on the empty cot in the prison cell. Gharl the imprisoned troll sleeps in Goth's bed.

This is always a problem with pathfinding the correct bed :(

> -After getting in a fight, sometimes the blood trails from the Avatar go on a long time, making a huge bloody mess everywhere, even if I fully heal myself.

We recently made Exult more bloody, seems we need to fine tune this more

**Reply by Zhelkas** [reply]
Yes, here's the savegame from right after Alagner gave me the key to his storeroom.  Unfortunately the next save I made was after I gave the notebook to the wisps, so I don't have a better one.

https://www.mediafire.com/file/e4csz6cqzovuzjf/exult48bg.sav/file

Thanks for looking into this.

**Reply by DominusExult** [reply]
@marzojr added some fix today for Alagner in BG Keyting. Might be fixing this issue (next automated snapshot in a few hours should have this).
And he also fixed healing with blood trail.

**Comment by Zhelkas** [comment]
Thanks!  I'll give it a shot.  Do you know if my save(s) after Alagner's "death" would be fixed?  Or should I go back to a save that's from before that point in the game?
________________________________

**Reply by DominusExult** [reply]
@Zhelkas I found no other way to contact you than here. take a look at https://ultimacodex.com/2023/10/check-out-dominus-dragons-re-creations-of-ultima-characters-using-hero-forge/ I've recreated the companions you did some time ago and have them actually printed and painted now. On the Ultima discord you can see pictures of these
